src/PixiEditor/Models/Controllers/MouseUpdateController.cs

```diff
@@ -1,49 +1,126 @@
-﻿using System.Timers;
+﻿using System.Diagnostics;
 using System.Windows;
 using System.Windows.Input;
 using System.Windows.Threading;
 
 namespace PixiEditor.Models.Controllers;
 
+#nullable enable
 public class MouseUpdateController : IDisposable
 {
-    private const int MouseUpdateIntervalMs = 7;  // 7ms ~= 142 Hz
-    
-    private readonly System.Timers.Timer _timer;
-    
-    private UIElement element;
-    
-    private MouseEventHandler mouseMoveHandler;
+    private const double MouseUpdateIntervalMs = 1000 / 142.0; //142 Hz
+
+    private Thread timerThread;
+    private readonly AutoResetEvent resetEvent = new(false);
+    private readonly object lockObj = new();
+    private bool isAborted = false;
+
+    private readonly FrameworkElement element;
+    private readonly MouseEventHandler mouseMoveHandler;
     
-    public MouseUpdateController(UIElement uiElement, MouseEventHandler onMouseMove)
+    public MouseUpdateController(FrameworkElement uiElement, MouseEventHandler onMouseMove)
     {
         mouseMoveHandler = onMouseMove;
         element = uiElement;
-        
-        _timer = new System.Timers.Timer(MouseUpdateIntervalMs);
-        _timer.Elapsed += TimerOnElapsed;
-        
         element.MouseMove += OnMouseMove;
+
+        bool wasThreadCreated = !element.IsLoaded;
+        element.Loaded += (_, _) =>
+        {
+            wasThreadCreated = true;
+            CreateTimerThread();
+        };
+
+        if (!wasThreadCreated)
+            CreateTimerThread();
+
+        element.Unloaded += (_, _) =>
+        {
+            isAborted = true;
+        };
+    }
+
+    private void CreateTimerThread()
+    {
+        timerThread = new Thread(TimerThread);
+        timerThread.Name = "MouseUpdateController thread";
+        timerThread.Start();
+        isAborted = false;
     }
 
-    private void TimerOnElapsed(object sender, ElapsedEventArgs e)
+    private bool IsThreadShouldStop()
+    {
+        return isAborted || timerThread != Thread.CurrentThread || Application.Current is null;
+    }
+    
+    private void TimerThread()
     {
-        Application.Current?.Dispatcher.Invoke(() =>
+        try
         {
-            element.MouseMove += OnMouseMove;
-        });
+            long lastThreadIter = Stopwatch.GetTimestamp();
+            
+            // abort if a new thread was created
+            while (!IsThreadShouldStop())
+            {
+                // call waitOne periodically instead of waiting infinitely to make sure we crash or exit when resetEvent is disposed
+                if (!resetEvent.WaitOne(300))
```

Comment:
You could make 300 a constant with a name, since I kinda don't know why its' 300